src/unitxt/loaders.py

```diff
@@ -39,13 +39,19 @@ class LoadHF(Loader):
     cached = False
 
     def process(self):
-        dataset = hf_load_dataset(
-            self.path, name=self.name, data_dir=self.data_dir, data_files=self.data_files, streaming=self.streaming
-        )
+        try:
```

Comment:
can't you just LoadHF with streaming=False?
and

if not streaming:
    for split in dataset.keys():
        dataset[split] = dataset[split].to_iterable_dataset()

?

Reply:
I think you can do as you suggest:
if (self.streaming):
    dataset = hf_load_dataset(
                self.path, name=self.name, data_dir=self.data_dir, data_files=self.data_files, streaming=self.streaming
            )
else:
      dataset = hf_load_dataset(
                self.path, name=self.name, data_dir=self.data_dir, data_files=self.data_files, streaming=False
            )
            for split in dataset.keys():
                dataset[split] = dataset[split].to_iterable_dataset()
                
Then the card creator , can have specify  

loader = LoadHF(path="GEM/xlsum", name="igbo", streaming=False) 


If the error message when stream = True and reading zip files is not clear, you can catch the exception, and raise a new one. 


In any case, best to have a test that checks that the exceptions is generated.

Reply:
Thank you, @yoavkatz , I do cover the exception catch in the test. I see these lines are all green in the coverage report.

Reply:
@yoavkatz , can you, then, approve this PR?